lib/fpm/package/gem.rb

```diff
@@ -47,6 +47,22 @@ class FPM::Package::Gem < FPM::Package
 
   option "--version-bins", :flag, "Append the version to the bins", :default => false
 
+  option "--stagingdir", "STAGINGDIR",
+    "The directory where fpm installs the gem temporarily before conversion. " \
+    "Normally a random subdirectory of workdir."
+
+  # Override parent method
+  def staging_path(path=nil)
+    @gem_staging_path ||= attributes[:gem_stagingdir] || Stud::Temporary.directory("package-#{type}-staging")
```

Comment:
Long-term I would prefer a simpler user experience for deterministic builds (perhaps a single flag `--deterministic-build`) that would make these kinds of decisions for the user. For now, this is good :)